### PR TITLE
[ci] Roll repo tooling and add readme-check

### PR DESCRIPTION
**Pull request by stuartmorgan**
- Rolls the pinned repo tooling to pick up the updated `readme-check`.
- Enables `readme-check` in CI, which I apparently forgot to ever do for this repository.
- Enables the secondary run with `--require-excerpts`, plus an opt-out list for all the current violations, so we can migrate incrementally.
- Fixes the few missing language tags so that the primary `readme-check` passes.
- Adds the submodule with the excerpt tools, and the task that runs excerpting, to allow for adopting excerpting in the future without repo-level changes.
  - As in flutter/plugins, cloning the submodule is not required for most use cases, so this should not cause disruption.

Brings flutter/packages in line with the recent flutter/plugins changes.

No version change: Adding language tags in READMEs is too minor to warrant releasing
No CHANGELOG change: Adding language tags in READMEs is too minor to warrant a mention

Part of https://github.com/flutter/flutter/issues/102679

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: script/configs/temp_exclude_excerpt.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Packages that have not yet adopted code-excerpt.
+#
+# This only exists to allow incrementally adopting the new requirement.
+# Packages shoud never be added to this list.
+
+# TODO(ecosystem): Remove everything from this list. See
```
Comment (Hixie) [inline script/configs/temp_exclude_excerpt.yaml:6]: fwiw generally for TODOs the name in brackets is the name of the person with context (not the name of the person responsible for fixing it)

**Comment by Hixie** [comment]
test-exempt: is a test.

File: script/configs/temp_exclude_excerpt.yaml (hunk 1 above)
Comment (stuartmorgan) [reply script/configs/temp_exclude_excerpt.yaml]: Fixed

**Review by ditman — APPROVED** [review]
LGTM! Let's get the snippets auto-extracted!

**Pull request merged**